## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 423602`

Logs: https://github.com/GaetanLepage/nixpkgs-review-gha/actions/runs/16153824114


---
### `x86_64-linux`
<details>
  <summary>:x: 40 packages failed to build:</summary>
  <ul>
    <li>python312Packages.pyls-flake8</li>
    <li>python312Packages.pyls-flake8.dist</li>
    <li>python312Packages.pyls-isort</li>
    <li>python312Packages.pyls-isort.dist</li>
    <li>python312Packages.pyls-memestra</li>
    <li>python312Packages.pyls-memestra.dist</li>
    <li>python312Packages.pyls-spyder</li>
    <li>python312Packages.pyls-spyder.dist</li>
    <li>python312Packages.pylsp-mypy</li>
    <li>python312Packages.pylsp-mypy.dist</li>
    <li>python312Packages.pylsp-rope</li>
    <li>python312Packages.pylsp-rope.dist</li>
    <li>python312Packages.python-lsp-black</li>
    <li>python312Packages.python-lsp-black.dist</li>
    <li>python312Packages.python-lsp-ruff</li>
    <li>python312Packages.python-lsp-ruff.dist</li>
    <li>python312Packages.python-lsp-server</li>
    <li>python312Packages.python-lsp-server.dist</li>
    <li>python312Packages.spyder</li>
    <li>python312Packages.spyder.dist</li>
    <li>python313Packages.pyls-flake8</li>
    <li>python313Packages.pyls-flake8.dist</li>
    <li>python313Packages.pyls-isort</li>
    <li>python313Packages.pyls-isort.dist</li>
    <li>python313Packages.pyls-memestra</li>
    <li>python313Packages.pyls-memestra.dist</li>
    <li>python313Packages.pyls-spyder</li>
    <li>python313Packages.pyls-spyder.dist</li>
    <li>python313Packages.pylsp-mypy</li>
    <li>python313Packages.pylsp-mypy.dist</li>
    <li>python313Packages.pylsp-rope</li>
    <li>python313Packages.pylsp-rope.dist</li>
    <li>python313Packages.python-lsp-black</li>
    <li>python313Packages.python-lsp-black.dist</li>
    <li>python313Packages.python-lsp-ruff</li>
    <li>python313Packages.python-lsp-ruff.dist</li>
    <li>python313Packages.python-lsp-server</li>
    <li>python313Packages.python-lsp-server.dist</li>
    <li>spyder (python313Packages.spyder)</li>
    <li>spyder.dist (python313Packages.spyder.dist)</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.python-lsp-server</summary>
<pre>plugins: flaky-3.8.1, cov-nixpkgs-stub-1.0.0
collecting ... 
collecting 144 items                                                           
collected 204 items / 1 error / 1 deselected / 203 selected                    

==================================== ERRORS ====================================
___________________ ERROR collecting test/test_python_lsp.py ___________________
ImportError while importing test module '/build/source/test/test_python_lsp.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/89laaz8x9pdydxzz74p0vd8vdj0szqbq-python3-3.12.11/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/test_python_lsp.py:11: in <module>
    import websockets
E   ModuleNotFoundError: No module named 'websockets'
----------------- generated xml file: /build/source/pytest.xml -----------------
=========================== short test summary info ============================
ERROR test/test_python_lsp.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
======================== 1 deselected, 1 error in 0.72s ========================</pre>
</details>
<details>
<summary>python313Packages.python-lsp-server</summary>
<pre>plugins: flaky-3.8.1, cov-nixpkgs-stub-1.0.0
collecting ... 
collecting 99 items                                                            
collected 204 items / 1 error / 1 deselected / 203 selected                    

==================================== ERRORS ====================================
___________________ ERROR collecting test/test_python_lsp.py ___________________
ImportError while importing test module '/build/source/test/test_python_lsp.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/sd81bvmch7njdpwx3lkjslixcbj5mivz-python3-3.13.4/lib/python3.13/importlib/__init__.py:88: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/test_python_lsp.py:11: in <module>
    import websockets
E   ModuleNotFoundError: No module named 'websockets'
----------------- generated xml file: /build/source/pytest.xml -----------------
=========================== short test summary info ============================
ERROR test/test_python_lsp.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
======================== 1 deselected, 1 error in 1.21s ========================</pre>
</details>
</details>


---
### `aarch64-linux`
<details>
  <summary>:x: 40 packages failed to build:</summary>
  <ul>
    <li>python312Packages.pyls-flake8</li>
    <li>python312Packages.pyls-flake8.dist</li>
    <li>python312Packages.pyls-isort</li>
    <li>python312Packages.pyls-isort.dist</li>
    <li>python312Packages.pyls-memestra</li>
    <li>python312Packages.pyls-memestra.dist</li>
    <li>python312Packages.pyls-spyder</li>
    <li>python312Packages.pyls-spyder.dist</li>
    <li>python312Packages.pylsp-mypy</li>
    <li>python312Packages.pylsp-mypy.dist</li>
    <li>python312Packages.pylsp-rope</li>
    <li>python312Packages.pylsp-rope.dist</li>
    <li>python312Packages.python-lsp-black</li>
    <li>python312Packages.python-lsp-black.dist</li>
    <li>python312Packages.python-lsp-ruff</li>
    <li>python312Packages.python-lsp-ruff.dist</li>
    <li>python312Packages.python-lsp-server</li>
    <li>python312Packages.python-lsp-server.dist</li>
    <li>python312Packages.spyder</li>
    <li>python312Packages.spyder.dist</li>
    <li>python313Packages.pyls-flake8</li>
    <li>python313Packages.pyls-flake8.dist</li>
    <li>python313Packages.pyls-isort</li>
    <li>python313Packages.pyls-isort.dist</li>
    <li>python313Packages.pyls-memestra</li>
    <li>python313Packages.pyls-memestra.dist</li>
    <li>python313Packages.pyls-spyder</li>
    <li>python313Packages.pyls-spyder.dist</li>
    <li>python313Packages.pylsp-mypy</li>
    <li>python313Packages.pylsp-mypy.dist</li>
    <li>python313Packages.pylsp-rope</li>
    <li>python313Packages.pylsp-rope.dist</li>
    <li>python313Packages.python-lsp-black</li>
    <li>python313Packages.python-lsp-black.dist</li>
    <li>python313Packages.python-lsp-ruff</li>
    <li>python313Packages.python-lsp-ruff.dist</li>
    <li>python313Packages.python-lsp-server</li>
    <li>python313Packages.python-lsp-server.dist</li>
    <li>spyder (python313Packages.spyder)</li>
    <li>spyder.dist (python313Packages.spyder.dist)</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>python312Packages.python-lsp-server</summary>
<pre>testpaths: test
plugins: flaky-3.8.1, cov-nixpkgs-stub-1.0.0
collecting ... 
collected 204 items / 1 error / 1 deselected / 203 selected                    

==================================== ERRORS ====================================
___________________ ERROR collecting test/test_python_lsp.py ___________________
ImportError while importing test module '/build/source/test/test_python_lsp.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/3ib0x9qs9bwg4nfkrk18rk1dx6yng0ga-python3-3.12.11/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/test_python_lsp.py:11: in <module>
    import websockets
E   ModuleNotFoundError: No module named 'websockets'
----------------- generated xml file: /build/source/pytest.xml -----------------
=========================== short test summary info ============================
ERROR test/test_python_lsp.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
======================== 1 deselected, 1 error in 0.56s ========================</pre>
</details>
<details>
<summary>python313Packages.python-lsp-server</summary>
<pre>testpaths: test
plugins: flaky-3.8.1, cov-nixpkgs-stub-1.0.0
collecting ... 
collected 204 items / 1 error / 1 deselected / 203 selected                    

==================================== ERRORS ====================================
___________________ ERROR collecting test/test_python_lsp.py ___________________
ImportError while importing test module '/build/source/test/test_python_lsp.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/avgafrfq80hx92p98m60wh2khmnmxrzc-python3-3.13.4/lib/python3.13/importlib/__init__.py:88: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/test_python_lsp.py:11: in <module>
    import websockets
E   ModuleNotFoundError: No module named 'websockets'
----------------- generated xml file: /build/source/pytest.xml -----------------
=========================== short test summary info ============================
ERROR test/test_python_lsp.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
======================== 1 deselected, 1 error in 0.55s ========================</pre>
</details>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 36 packages failed to build:</summary>
  <ul>
    <li>python312Packages.pyls-flake8</li>
    <li>python312Packages.pyls-flake8.dist</li>
    <li>python312Packages.pyls-isort</li>
    <li>python312Packages.pyls-isort.dist</li>
    <li>python312Packages.pyls-memestra</li>
    <li>python312Packages.pyls-memestra.dist</li>
    <li>python312Packages.pyls-spyder</li>
    <li>python312Packages.pyls-spyder.dist</li>
    <li>python312Packages.pylsp-mypy</li>
    <li>python312Packages.pylsp-mypy.dist</li>
    <li>python312Packages.pylsp-rope</li>
    <li>python312Packages.pylsp-rope.dist</li>
    <li>python312Packages.python-lsp-black</li>
    <li>python312Packages.python-lsp-black.dist</li>
    <li>python312Packages.python-lsp-ruff</li>
    <li>python312Packages.python-lsp-ruff.dist</li>
    <li>python312Packages.python-lsp-server</li>
    <li>python312Packages.python-lsp-server.dist</li>
    <li>python313Packages.pyls-flake8</li>
    <li>python313Packages.pyls-flake8.dist</li>
    <li>python313Packages.pyls-isort</li>
    <li>python313Packages.pyls-isort.dist</li>
    <li>python313Packages.pyls-memestra</li>
    <li>python313Packages.pyls-memestra.dist</li>
    <li>python313Packages.pyls-spyder</li>
    <li>python313Packages.pyls-spyder.dist</li>
    <li>python313Packages.pylsp-mypy</li>
    <li>python313Packages.pylsp-mypy.dist</li>
    <li>python313Packages.pylsp-rope</li>
    <li>python313Packages.pylsp-rope.dist</li>
    <li>python313Packages.python-lsp-black</li>
    <li>python313Packages.python-lsp-black.dist</li>
    <li>python313Packages.python-lsp-ruff</li>
    <li>python313Packages.python-lsp-ruff.dist</li>
    <li>python313Packages.python-lsp-server</li>
    <li>python313Packages.python-lsp-server.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>python312Packages.python-lsp-server</summary>
<pre>collecting 5 items                                                             
collecting 99 items                                                            
collecting 155 items / 1 error                                                 
collected 204 items / 1 error / 1 deselected / 203 selected                    

==================================== ERRORS ====================================
___________________ ERROR collecting test/test_python_lsp.py ___________________
ImportError while importing test module '/nix/build/nix-build-python3.12-python-lsp-server-1.13.0.drv-0/source/test/test_python_lsp.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/q2y1zwszjmsap7r6nbk4dxbcsm47abfx-python3-3.12.11/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/test_python_lsp.py:11: in <module>
    import websockets
E   ModuleNotFoundError: No module named 'websockets'
- generated xml file: /nix/build/nix-build-python3.12-python-lsp-server-1.13.0.drv-0/source/pytest.xml -
=========================== short test summary info ============================
ERROR test/test_python_lsp.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
======================== 1 deselected, 1 error in 2.32s ========================</pre>
</details>
<details>
<summary>python313Packages.python-lsp-server</summary>
<pre>collecting 5 items                                                             
collecting 99 items                                                            
collecting 159 items / 1 error                                                 
collected 204 items / 1 error / 1 deselected / 203 selected                    

==================================== ERRORS ====================================
___________________ ERROR collecting test/test_python_lsp.py ___________________
ImportError while importing test module '/nix/build/nix-build-python3.13-python-lsp-server-1.13.0.drv-0/source/test/test_python_lsp.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/il3s6rjh49l9ccg1nlqm93w67lcnfmqn-python3-3.13.4/lib/python3.13/importlib/__init__.py:88: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/test_python_lsp.py:11: in <module>
    import websockets
E   ModuleNotFoundError: No module named 'websockets'
- generated xml file: /nix/build/nix-build-python3.13-python-lsp-server-1.13.0.drv-0/source/pytest.xml -
=========================== short test summary info ============================
ERROR test/test_python_lsp.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
======================== 1 deselected, 1 error in 2.16s ========================</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 36 packages failed to build:</summary>
  <ul>
    <li>python312Packages.pyls-flake8</li>
    <li>python312Packages.pyls-flake8.dist</li>
    <li>python312Packages.pyls-isort</li>
    <li>python312Packages.pyls-isort.dist</li>
    <li>python312Packages.pyls-memestra</li>
    <li>python312Packages.pyls-memestra.dist</li>
    <li>python312Packages.pyls-spyder</li>
    <li>python312Packages.pyls-spyder.dist</li>
    <li>python312Packages.pylsp-mypy</li>
    <li>python312Packages.pylsp-mypy.dist</li>
    <li>python312Packages.pylsp-rope</li>
    <li>python312Packages.pylsp-rope.dist</li>
    <li>python312Packages.python-lsp-black</li>
    <li>python312Packages.python-lsp-black.dist</li>
    <li>python312Packages.python-lsp-ruff</li>
    <li>python312Packages.python-lsp-ruff.dist</li>
    <li>python312Packages.python-lsp-server</li>
    <li>python312Packages.python-lsp-server.dist</li>
    <li>python313Packages.pyls-flake8</li>
    <li>python313Packages.pyls-flake8.dist</li>
    <li>python313Packages.pyls-isort</li>
    <li>python313Packages.pyls-isort.dist</li>
    <li>python313Packages.pyls-memestra</li>
    <li>python313Packages.pyls-memestra.dist</li>
    <li>python313Packages.pyls-spyder</li>
    <li>python313Packages.pyls-spyder.dist</li>
    <li>python313Packages.pylsp-mypy</li>
    <li>python313Packages.pylsp-mypy.dist</li>
    <li>python313Packages.pylsp-rope</li>
    <li>python313Packages.pylsp-rope.dist</li>
    <li>python313Packages.python-lsp-black</li>
    <li>python313Packages.python-lsp-black.dist</li>
    <li>python313Packages.python-lsp-ruff</li>
    <li>python313Packages.python-lsp-ruff.dist</li>
    <li>python313Packages.python-lsp-server</li>
    <li>python313Packages.python-lsp-server.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python312Packages.python-lsp-server</summary>
<pre>plugins: flaky-3.8.1, cov-nixpkgs-stub-1.0.0
collecting ... 
collecting 204 items / 1 error                                                 
collected 204 items / 1 error / 1 deselected / 203 selected                    

==================================== ERRORS ====================================
___________________ ERROR collecting test/test_python_lsp.py ___________________
ImportError while importing test module '/nix/build/nix-build-python3.12-python-lsp-server-1.13.0.drv-0/source/test/test_python_lsp.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/q54d9qqlb27g2nf9ja4zsa89gl7b3yvm-python3-3.12.11/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/test_python_lsp.py:11: in <module>
    import websockets
E   ModuleNotFoundError: No module named 'websockets'
- generated xml file: /nix/build/nix-build-python3.12-python-lsp-server-1.13.0.drv-0/source/pytest.xml -
=========================== short test summary info ============================
ERROR test/test_python_lsp.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
======================== 1 deselected, 1 error in 0.57s ========================</pre>
</details>
<details>
<summary>python313Packages.python-lsp-server</summary>
<pre>plugins: flaky-3.8.1, cov-nixpkgs-stub-1.0.0
collecting ... 
collecting 204 items / 1 error                                                 
collected 204 items / 1 error / 1 deselected / 203 selected                    

==================================== ERRORS ====================================
___________________ ERROR collecting test/test_python_lsp.py ___________________
ImportError while importing test module '/nix/build/nix-build-python3.13-python-lsp-server-1.13.0.drv-0/source/test/test_python_lsp.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/j1spz18w5a10r9nj4wpkyh0cx730rs4z-python3-3.13.4/lib/python3.13/importlib/__init__.py:88: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/test_python_lsp.py:11: in <module>
    import websockets
E   ModuleNotFoundError: No module named 'websockets'
- generated xml file: /nix/build/nix-build-python3.13-python-lsp-server-1.13.0.drv-0/source/pytest.xml -
=========================== short test summary info ============================
ERROR test/test_python_lsp.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
======================== 1 deselected, 1 error in 0.59s ========================</pre>
</details>
</details>

--- a/pkgs/development/python-modules/python-lsp-server/default.nix
+++ b/pkgs/development/python-modules/python-lsp-server/default.nix
@@ -7,6 +7,7 @@
   setuptools-scm,
 
   # dependencies
+  black,
   docstring-to-markdown,
   jedi,
   pluggy,
@@ -41,14 +42,14 @@
 
 buildPythonPackage rec {
   pname = "python-lsp-server";
-  version = "1.12.2";
+  version = "1.13.0";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "python-lsp";
     repo = "python-lsp-server";
     tag = "v${version}";
-    hash = "sha256-tdhYLAXs1Yf3DqCzf/pLOlJvr/zYRkSlAF6hsavSu+A=";
+    hash = "sha256-NIqBIB4IG4xo7zDhaafWvT1RUnkqup1gm9WQhdtpIfc=";
   };
 
   pythonRelaxDeps = [
@@ -63,6 +64,7 @@ buildPythonPackage rec {
   build-system = [ setuptools-scm ];
 
   dependencies = [
+    black
     docstring-to-markdown
     jedi
     pluggy
@@ -82,6 +84,7 @@ buildPythonPackage rec {
       pylint
       rope
       toml
+      websockets
       whatthepatch
       yapf
     ];
@@ -116,6 +119,9 @@ buildPythonPackage rec {
   disabledTests = [
     # avoid dependencies on many Qt things just to run one singular test
     "test_pyqt_completion"
+
+    # Flaky: ValueError: I/O operation on closed file
+    "test_concurrent_ws_requests"
   ];
 
   pythonImportsCheck = [

--- a/pkgs/development/python-modules/spyder/default.nix
+++ b/pkgs/development/python-modules/spyder/default.nix
@@ -74,6 +74,7 @@ buildPythonPackage rec {
 
   pythonRelaxDeps = [
     "ipython"
+    "python-lsp-server"
   ];
 
   dependencies = [

